editor/src/clj/editor/scene.clj

```diff
@@ -1051,21 +1049,74 @@
                          :perspective (c/camera-orthographic->perspective old-camera c/fov-y-35mm-full-frame))]
         (set-camera! camera-controller old-camera new-camera false)))))
 
+(defn- sync-camera-position
+  [^Camera camera-a ^Camera camera-b viewport]
+  (let [focus ^Vector4d (:focus-point camera-b)
+        point (c/camera-project camera-b viewport (Point3d. (.x focus) (.y focus) (.z focus)))
+        world (c/camera-unproject camera-a viewport (.x point) (.y point) (.z point))
+        delta (c/camera-unproject camera-b viewport (.x point) (.y point) (.z point))]
+    (.sub delta world)
+    (c/camera-move camera-a (.x delta) (.y delta) (.z delta))))
```

Comment:
When we navigate on 2d and then switch to 3d, we attempt to move the camera to the updated location, but keep the rotation of the cached 3d camera. Not sure if that's the best way to handle this.